File: lib/tauri/schedule.ts
import { invoke } from '@tauri-apps/api/core'

import { type Schedule } from '@/types/schedule'

export function getSchedule() {
  return invoke<Schedule>('get_schedule')
}

export function updateSchedule(schedule: Schedule) {
  return invoke<Schedule>('update_schedule', { input: schedule })
}

File: src-tauri/src/commands/schedule.rs
use tauri::State;

use crate::db::models::{Schedule, ScheduleInput};
use crate::db::{DbState, repo, run};

#[tauri::command]
pub async fn get_schedule(state: State<'_, DbState>) -> Result<Schedule, String> {
    run(state, repo::get_schedule).await
}

#[tauri::command]
pub async fn update_schedule(
    state: State<'_, DbState>,
    input: ScheduleInput,
) -> Result<Schedule, String> {
    run(state, move |conn| repo::update_schedule(conn, input)).await
}
